### PR TITLE
[COM-1513] Export Chakra Drawer

### PR DESCRIPTION
**Pull request by Rogerpeke97**
Add Chakra Drawer Export

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Storybook or docs have been added / updated (for bug fixes / features)

[COM-1513](https://impulsumvc.atlassian.net/browse/COM-1513?atlOrigin=eyJpIjoiMGIzNzc2Y2Y0NTY0NDYzZDhkOGE1MzRlOTZiNmQ2MjEiLCJwIjoiaiJ9)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)

[COM-1513]: https://impulsumvc.atlassian.net/browse/COM-1513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/components/overlay/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@ export * from './Dropdown';
 export * from './Modal';
 export * from './Popover';
 export * from './Tooltip';
+export * from './Drawer';
```
Comment (guilean) [inline packages/components/src/components/overlay/index.ts:5]: nit: could we order alphabetically?

File: packages/components/src/components/overlay/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const DrawerBody: FC<ModalBodyProps> = ({ children, ...rest }) => (
+  <ChakraDrawerBody {...rest}>
+    {children}
+  </ChakraDrawerBody>
+);
+
+const DrawerCloseButton = forwardRef<CloseButtonProps, typeof CloseButton>((props, ref) => {
+  const { onClick, ...rest } = props;
+  const { onClose } = useModalContext();
```
Comment (guilean) [inline packages/components/src/components/overlay/Drawer/Drawer.tsx:52]: Can we use the `useDisclosure` hook instead?

File: packages/components/src/components/overlay/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+export const Drawer: FC<DrawerProps> & DrawerStaticMembers = ({ children, ...rest }) => (
+  <ChakraDrawer {...rest}>
+    {children}
+  </ChakraDrawer>
+);
+
+const DrawerOverlay: FC<ModalOverlayProps> = (props) => <ChakraDrawerOverlay {...props} />;
+
+const DrawerContent: FC<DrawerContentProps> = ({ children, ...rest }) => (
```
Comment (guilean) [inline packages/components/src/components/overlay/Drawer/Drawer.tsx:31]: Could we simplify each static member that doesn't have any overrides to:

`Drawer.Content = ChakraDrawerContent`
...

**Review by guilean — COMMENTED** [review]
Remember that you can generate a canary build by adding the label in the PR. This way you can test the build wherever you need

File: packages/components/src/components/overlay/index.ts (hunk 1 above)
Comment (Rogerpeke97) [reply packages/components/src/components/overlay/index.ts]: Done

File: packages/components/src/components/overlay/Drawer/Drawer.tsx (hunk 2 above)
Comment (Rogerpeke97) [reply packages/components/src/components/overlay/Drawer/Drawer.tsx:26]: Didn't work :/

File: packages/components/src/components/overlay/Drawer/Drawer.tsx (hunk 3 above)
Comment (Rogerpeke97) [reply packages/components/src/components/overlay/Drawer/Drawer.tsx]: Done 👍🏼

File: packages/components/src/components/overlay/Drawer/Drawer.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { DrawerStaticMembers } from './types';
+import { CloseButton, CloseButtonProps } from '@components';
+import { callAllHandlers } from '@chakra-ui/utils';
+
+
+export const Drawer: FC<DrawerProps> & DrawerStaticMembers = ({ children, ...rest }) => (
+  <ChakraDrawer {...rest}>
+    {children}
+  </ChakraDrawer>
+);
```
Comment (guilean) [inline packages/components/src/components/overlay/Drawer/Drawer.tsx:22]:
```suggestion
export const Drawer: FC<DrawerProps> & DrawerStaticMembers = (props) => <ChakraDrawer {...props} />;
```

**Review by guilean — APPROVED** [review]

File: packages/components/src/components/overlay/Drawer/Drawer.test.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    screen.getByText(/Drawer Header Children/i);
+  });
+
+  test('should not render child if no open', () => {
+    givenComponentRendered(false);
+    expect(screen.queryByText(/Drawer Body Children/i)).not.toBeInTheDocument();
+    expect(screen.queryByText(/Drawer Header Children/i)).not.toBeInTheDocument();
+  });
+});
+
```
Comment (tkanzakic) [inline packages/components/src/components/overlay/Drawer/Drawer.test.tsx:37]: nit: extra line at the end of the file

File: packages/components/src/components/overlay/Drawer/Drawer.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  );
+});
+
+Drawer.Overlay = ChakraDrawerOverlay;
+Drawer.Content = ChakraDrawerContent;
+Drawer.Footer = ChakraDrawerFooter;
+Drawer.Body = ChakraDrawerBody;
+Drawer.CloseButton = DrawerCloseButton;
+Drawer.Header = ChakraDrawerHeader;
+
```
Comment (tkanzakic) [inline packages/components/src/components/overlay/Drawer/Drawer.tsx:46]: ditto

File: packages/components/src/components/overlay/Drawer/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export * from './Drawer';
+export * from './types';
+
```
Comment (tkanzakic) [inline packages/components/src/components/overlay/Drawer/index.ts:3]: ditto

File: packages/components/src/components/overlay/Drawer/types.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+export interface DrawerStaticMembers {
+  Overlay: FC<ModalOverlayProps>;
+  Content: FC<DrawerContentProps>;
+  Footer: FC<ModalFooterProps>;
+  Body: FC<ModalBodyProps>;
+  CloseButton: FC<CloseButtonProps>;
+  Header: FC<ModalHeaderProps>;
+}
+
+
```
Comment (tkanzakic) [inline packages/components/src/components/overlay/Drawer/types.ts:20]: ditto

**Pull request merged**